pkg/sync/sync_context.go

```diff
@@ -792,7 +794,10 @@ func (sc *syncContext) autoCreateNamespace(tasks syncTasks) syncTasks {
 					}
 				}
 			} else if apierr.IsNotFound(err) {
-				tasks = append(tasks, &syncTask{phase: common.SyncPhasePreSync, targetObj: unstructuredObj, liveObj: nil})
+				unstructuredObjCopy := unstructuredObj.DeepCopy()
+				if sc.namespaceCreator(unstructuredObjCopy) {
+					tasks = append(tasks, &syncTask{phase: common.SyncPhasePreSync, targetObj: unstructuredObjCopy, liveObj: nil})
+				}
```

Comment:
Is there a good reason for using a copy of `unstructuredObj` here instead of just using the original?

Reply:
it is in same way for modifier, i decided to do in same way here for prevent issues, probably no 
https://github.com/argoproj/gitops-engine/pull/443/files#diff-74e77a4722803a7d3f87b956ec81b82d3c5a0dd50de299922aa2208fec57e6e9R791

Reply:
I think we can remove copy function but in such case we need creator func should apply unstructuredObj  as value and not reference,

Reply:
yes, it can be removed